packages/vite-plugin-cloudflare/src/runner/worker.ts

```diff
@@ -69,50 +70,120 @@ function createModuleRunner(env: RunnerEnv, webSocket: WebSocket) {
 let moduleRunner: ModuleRunner;
 let entrypoint: string;
 
-export default {
-	async fetch(request, env, ctx) {
-		const url = new URL(request.url);
+export default class extends WorkerEntrypoint<RunnerEnv> {
+	constructor(ctx: ExecutionContext, env: RunnerEnv) {
+		super(ctx, env);
+
+		return new Proxy(this, {
+			get(target, prop) {
+				if (prop === 'fetch') {
+					return async (request: Request) => {
+						const url = new URL(request.url);
+
+						if (url.pathname === INIT_PATH) {
+							return target.#init.apply(target, [request]);
+						}
+
+						return target.#fetch.apply(target, [request]);
+					};
+				} else {
+					return (...args: unknown[]) => {
+						return target.#rpc.apply(target, [prop, args]);
```

Comment:
this assumes that all the properties of a `WorkerEntrypoint` have to be methods, is that tha case? can't I simply expose a constant from my class instance?

Reply:
Good point. I thought you could only call methods over RPC but it turns out you can also access getters - https://developers.cloudflare.com/workers/runtime-apis/rpc/visibility/#class-instance-properties. I'll add support for this.